ValidationError: Row #1: Selling rate for item <strong>_Test Item for Drop Shipping Selling <strong>net rate</strong> should be atleast <strong>188700.0</strong>.<br><br>Alternatively, you can disable selling price validation in Selling Settings

### Information about bug

Below mentioned 48 TCs are failing
<google-sheets-html-origin><style type="text/css"><!--td {border: 1px solid #cccccc;}br {mso-data-placement:same-cell;}--></style>
Test Case ID | Issue Type Recent
-- | --
TC_S_112 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item for Drop Shipping</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>188700.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_114 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item for Drop Shipping</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>188700.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_116 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item for Drop Shipping</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>188700.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_032 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item for quality inspection</strong> is lower than its <strong>valuation rate (Moving Average)</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_110 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>210000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_108 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>210000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_136 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>210000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_137 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>210000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_113 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>210000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_115 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>210000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_153 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_SCK_143 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_047 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_046 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_185 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_180 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_171 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_172 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_179 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_169 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_164 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_178 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_186 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_190 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_177 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_175 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_035 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_037 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_033 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_036 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_034 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_040 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_127 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_049 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_192 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_182 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_ACC_101 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>690.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_158 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>690.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_159 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>690.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_129 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>690.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_128 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>690.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_130 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>690.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_131 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>690.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_139 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.					Selling <strong>net rate</strong> should be atleast <strong>690.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_140 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>valuation rate (Moving Average)</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_141 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>valuation rate (Moving Average)</strong>.					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_S_135 | raise excfrappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>valuation rate (Moving Average)</strong>.					Selling <strong>net rate</strong> should be atleast <strong>5000.0</strong>.<br><br>Alternatively,					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass					this validation.
TC_ACC_126 | + Cannot delete or cancel because Sales Invoice None is linked with Payment Entry None at Row: 1- Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.- 					Selling <strong>net rate</strong> should be atleast <strong>690.0</strong>.<br><br>Alternatively,- 					you can disable selling price validation in <a href="http://test_site:8000/app/selling-settings/Selling%20Settings">Selling Settings</a> to bypass- 					this validation.



### Module

other

### Version

Assets: v0.0.1 (pre-prod)

ERPNext: v15.58.2 (pre-prod)

ERPNext CRM: v0.0.1 (pre-prod)

ERPNext India: v0.0.1 (pre-prod)

Frappe Framework: v15.63.1 (pre-prod)

India Compliance: v15.17.1 (pre-prod)

Payments: v0.0.1 (version-15)

Projects: v0.0.1 (pre-prod)

Sales Commission: v0.0.1 (pre-prod)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell
"Traceback (most recent call last):
  File ""/home/runner/frappe-bench/apps/erpnext/erpnext/selling/doctype/sales_order/test_sales_order.py"", line 6393, in test_check_modified_date_coverage_TC_S_164
    so.save()
  File ""/home/runner/frappe-bench/apps/frappe/frappe/model/document.py"", line 381, in save
    return self._save(*args, **kwargs)
  File ""/home/runner/frappe-bench/apps/frappe/frappe/model/document.py"", line 403, in _save
    return self.insert()
  File ""/home/runner/frappe-bench/apps/frappe/frappe/model/document.py"", line 312, in insert
    self.run_before_save_methods()
  File ""/home/runner/frappe-bench/apps/frappe/frappe/model/document.py"", line 1140, in run_before_save_methods
    self.run_method(""validate"")
  File ""/home/runner/frappe-bench/apps/frappe/frappe/model/document.py"", line 1011, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File ""/home/runner/frappe-bench/apps/frappe/frappe/model/document.py"", line 1371, in composer
    return composed(self, method, *args, **kwargs)
  File ""/home/runner/frappe-bench/apps/frappe/frappe/model/document.py"", line 1353, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File ""/home/runner/frappe-bench/apps/frappe/frappe/model/document.py"", line 1008, in fn
    return method_object(*args, **kwargs)
  File ""/home/runner/frappe-bench/apps/erpnext/erpnext/selling/doctype/sales_order/sales_order.py"", line 173, in validate
    super().validate()
  File ""/home/runner/frappe-bench/apps/erpnext/erpnext/controllers/selling_controller.py"", line 38, in validate
    self.validate_selling_price()
  File ""/home/runner/frappe-bench/apps/erpnext/erpnext/controllers/selling_controller.py"", line 263, in validate_selling_price
    throw_message(item.idx, item.item_name, last_purchase_rate_in_sales_uom, ""last purchase rate"")
  File ""/home/runner/frappe-bench/apps/erpnext/erpnext/controllers/selling_controller.py"", line 227, in throw_message
    throw(
  File ""/home/runner/frappe-bench/apps/frappe/frappe/__init__.py"", line 634, in throw
    msgprint(
  File ""/home/runner/frappe-bench/apps/frappe/frappe/__init__.py"", line 599, in msgprint
    _raise_exception()
  File ""/home/runner/frappe-bench/apps/frappe/frappe/__init__.py"", line 550, in _raise_exception
    raise exc
frappe.exceptions.ValidationError: Row #1: Selling rate for item <strong>_Test Item</strong> is lower than its <strong>last purchase rate</strong>.
					Selling <strong>net rate</strong> should be atleast <strong>500.0</strong>.<br><br>Alternatively,
					you can disable selling price validation in <a href=""http://test_site:8000/app/selling-settings/Selling%20Settings"">Selling Settings</a> to bypass
					this validation."
```

@ashishverma8848 , @anandk-spyke-o1  Getting issues in postgres instance
TC_S_046 - AssertionError: None != 900
TC_S_047 - AssertionError: None != 500.0

TC_S_035 -
AssertionError: 'Paid' != 'Unpaid'
- Paid
+ Unpaid
 : Sales Invoice not created


TC_S_139 - AssertionError: 400.0 != 1000

--- a/erpnext/accounts/doctype/pos_invoice/test_pos_invoice.py
+++ b/erpnext/accounts/doctype/pos_invoice/test_pos_invoice.py
@@ -744,7 +744,7 @@ class TestPOSInvoice(unittest.TestCase):
 		rounded_total = frappe.db.get_value("Sales Invoice", pos_inv.consolidated_invoice, "rounded_total")
 		self.assertEqual(rounded_total, 840)
 
-	@change_settings("Selling Settings",{"validate_selling_price":1})
+
 	def test_merging_with_validate_selling_price(self):
 		from erpnext.accounts.doctype.pos_closing_entry.test_pos_closing_entry import (
 			init_user_and_profile,
@@ -753,6 +753,8 @@ class TestPOSInvoice(unittest.TestCase):
 			consolidate_pos_invoices,
 		)
 
+		if not frappe.db.get_single_value("Selling Settings", "validate_selling_price"):
+			frappe.db.set_single_value("Selling Settings", "validate_selling_price", 1)
 		item = "Test Selling Price Validation"
 		make_item(item, {"is_stock_item": 1})
 		make_purchase_receipt(item_code=item, warehouse="_Test Warehouse - _TC", qty=1, rate=300)
@@ -789,14 +791,14 @@ class TestPOSInvoice(unittest.TestCase):
 		pos_inv2.save()
 		pos_inv2.submit()
 
+		frappe.db.set_single_value("Selling Settings", "validate_selling_price", 0)
+
 		consolidate_pos_invoices()
 
 		pos_inv2.load_from_db()
 		rounded_total = frappe.db.get_value("Sales Invoice", pos_inv2.consolidated_invoice, "rounded_total")
 		self.assertEqual(rounded_total, 400)
-		frappe.clear_cache(doctype="Selling Settings")
 
-	@change_settings("Selling Settings",{"validate_selling_price":0})
 	def test_pos_batch_reservation(self):
 		from erpnext.stock.doctype.serial_and_batch_bundle.serial_and_batch_bundle import (
 			get_auto_batch_nos,
@@ -856,7 +858,6 @@ class TestPOSInvoice(unittest.TestCase):
 		for batch in batches:
 			if batch.batch_no == batch_no and batch.warehouse == "_Test Warehouse - _TC":
 				self.assertEqual(batch.qty, 5)
-		frappe.clear_cache(doctype="Selling Settings")
 
 	def test_pos_batch_item_qty_validation(self):
 		from erpnext.stock.doctype.serial_and_batch_bundle.serial_and_batch_bundle import (

--- a/erpnext/accounts/doctype/pricing_rule/test_pricing_rule.py
+++ b/erpnext/accounts/doctype/pricing_rule/test_pricing_rule.py
@@ -15,6 +15,8 @@ from erpnext.stock.doctype.stock_entry.stock_entry_utils import make_stock_entry
 
 class TestPricingRule(FrappeTestCase):
 	def setUp(self):
+		if frappe.db.get_single_value("Selling Settings", "validate_selling_price"):
+			frappe.db.set_single_value("Selling Settings", "validate_selling_price", 0)
 		delete_existing_pricing_rules()
 		if "custom_crm" in frappe.get_installed_apps():
 			setup_pricing_rule_data()
@@ -28,6 +30,8 @@ class TestPricingRule(FrappeTestCase):
 
 	def tearDown(self):
 		delete_existing_pricing_rules()
+		if frappe.db.get_single_value("Selling Settings", "validate_selling_price"):
+			frappe.db.set_single_value("Selling Settings", "validate_selling_price", 0)
 
 	def test_pricing_rule_for_discount(self):
 		from frappe import MandatoryError
@@ -435,52 +439,52 @@ class TestPricingRule(FrappeTestCase):
 		self.assertEqual(so.items[1].item_code, "_Test Item 2")
 
 	def test_enforce_free_item_qty(self):
- 		# this test is only for testing non-enforcement as all other tests in this file already test with enforcement
- 		frappe.delete_doc_if_exists("Pricing Rule", "_Test Pricing Rule")
- 		test_record = {
- 			"doctype": "Pricing Rule",
- 			"title": "_Test Pricing Rule",
- 			"apply_on": "Item Code",
- 			"currency": "USD",
- 			"items": [
- 				{
- 					"item_code": "_Test Item",
- 				}
- 			],
- 			"selling": 1,
- 			"rate_or_discount": "Discount Percentage",
- 			"rate": 0,
- 			"min_qty": 0,
- 			"max_qty": 7,
- 			"discount_percentage": 17.5,
- 			"price_or_product_discount": "Product",
- 			"same_item": 0,
- 			"free_item": "_Test Item 2",
- 			"free_qty": 1,
- 			"company": "_Test Company",
- 		}
- 		pricing_rule = frappe.get_doc(test_record.copy()).insert()
- 
- 		# With enforcement
- 		so = make_sales_order(item_code="_Test Item", qty=1, do_not_submit=True)
- 		self.assertEqual(so.items[1].is_free_item, 1)
- 		self.assertEqual(so.items[1].item_code, "_Test Item 2")
- 
- 		# Test 1 : Saving a document with an item with pricing list without it's corresponding free item will cause it the free item to be refetched on save
- 		so.items.pop(1)
- 		so.save()
- 		so.reload()
- 		self.assertEqual(len(so.items), 2)
- 
- 		# Without enforcement
- 		pricing_rule.enforce_free_item_qty = 0
- 		pricing_rule.save()
- 
- 		# Test 2 : Deleted free item will not be fetched again on save without enfrocement
- 		so.items.pop(1)
- 		so.save()
- 		so.reload()
- 		self.assertEqual(len(so.items), 1)
+		# this test is only for testing non-enforcement as all other tests in this file already test with enforcement
+		frappe.delete_doc_if_exists("Pricing Rule", "_Test Pricing Rule")
+		test_record = {
+			"doctype": "Pricing Rule",
+			"title": "_Test Pricing Rule",
+			"apply_on": "Item Code",
+			"currency": "USD",
+			"items": [
+				{
+					"item_code": "_Test Item",
+				}
+			],
+			"selling": 1,
+			"rate_or_discount": "Discount Percentage",
+			"rate": 0,
+			"min_qty": 0,
+			"max_qty": 7,
+			"discount_percentage": 17.5,
+			"price_or_product_discount": "Product",
+			"same_item": 0,
+			"free_item": "_Test Item 2",
+			"free_qty": 1,
+			"company": "_Test Company",
+		}
+		pricing_rule = frappe.get_doc(test_record.copy()).insert()
+
+		# With enforcement
+		so = make_sales_order(item_code="_Test Item", qty=1, do_not_submit=True)
+		self.assertEqual(so.items[1].is_free_item, 1)
+		self.assertEqual(so.items[1].item_code, "_Test Item 2")
+
+		# Test 1 : Saving a document with an item with pricing list without it's corresponding free item will cause it the free item to be refetched on save
+		so.items.pop(1)
+		so.save()
+		so.reload()
+		self.assertEqual(len(so.items), 2)
+
+		# Without enforcement
+		pricing_rule.enforce_free_item_qty = 0
+		pricing_rule.save()
+
+		# Test 2 : Deleted free item will not be fetched again on save without enfrocement
+		so.items.pop(1)
+		so.save()
+		so.reload()
+		self.assertEqual(len(so.items), 1)
 
 	def test_dont_enforce_free_item_qty(self):
 		# this test is only for testing non-enforcement as all other tests in this file already test with enforcement

--- a/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
+++ b/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
@@ -5728,7 +5728,7 @@ class TestSalesInvoice(FrappeTestCase):
 		]
 		check_gl_entries(self, jv_doc.name, expected_gl_entries, jv_doc.posting_date, "Journal Entry")
 	
-	@change_settings("Selling Settings",{"validate_selling_price":1})
+
 	def test_prevent_sale_below_purchase_rate_TC_ACC_125(self):
 		from erpnext.accounts.doctype.payment_entry.test_payment_entry import (
 			create_purchase_invoice,
@@ -5740,6 +5740,9 @@ class TestSalesInvoice(FrappeTestCase):
 		import frappe
 
 		get_or_create_fiscal_year("_Test Company")
+
+		if not frappe.db.get_single_value("Selling Settings", "validate_selling_price"):
+			frappe.db.set_single_value("Selling Settings", "validate_selling_price", 1)
 
 		if not frappe.db.get_value("Company","_Test Company","stock_received_but_not_billed"):
 			frappe.db.set_value("Company","_Test Company","stock_received_but_not_billed","Stock Received But Not Billed - _TC")
@@ -5773,9 +5776,10 @@ class TestSalesInvoice(FrappeTestCase):
 		with self.assertRaises(frappe.ValidationError) as context:
 			si.save()
 		self.assertEqual(str(context.exception), expected_msg)
-		frappe.clear_cache(doctype="Selling Settings")
+
+		frappe.db.set_single_value("Selling Settings", "validate_selling_price", 0)
 	
-	@change_settings("Selling Settings",{"validate_selling_price":0})
+
 	def test_test_unlink_payment_on_invoice_cancellation_TC_ACC_126(self):
 		from erpnext.accounts.doctype.payment_entry.test_payment_entry import (
 			make_test_item
@@ -5806,8 +5810,7 @@ class TestSalesInvoice(FrappeTestCase):
 		except Exception as e:
 			error_msg = str(e)
 			self.assertEqual(error_msg,f'Cannot delete or cancel because Sales Invoice {si_name} is linked with Payment Entry {pe_name} at Row: 1')
-		finally:
-			frappe.clear_cache(doctype="Selling Settings")
+
 
 	def test_si_cancel_amend_with_item_details_change_TC_S_128(self):
 		from erpnext.accounts.doctype.payment_entry.test_payment_entry import (

